File: C++ solutions/1748. Sum of Unique Elements.cpp
// approach :1
class Solution {
public:
    int sumOfUnique(vector<int>& nums) {
        map<int,int>m;
        int ans=0;
        for(int i=0;i<nums.size();i++) m[nums[i]]++;
        for(auto k:m) if(k.second==1) ans+=k.first;
        return ans;
    }
};

// approach :2
class Solution {
public:
    int sumOfUnique(vector<int>& nums) {
        map<int,int>m;
        int ans=0;
        for(int i=0;i<nums.size();i++){
            m[nums[i]]++;
            if(m[nums[i]]>1){
                if(m[nums[i]]==2) ans-=nums[i];
            }
            else ans+=nums[i];
        }
        return ans;
    }
};

File: JavaScript solutions/1748. Sum of Unique Elements.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var sumOfUnique = function (nums) {
  const map = new Map();
  let ans = 0;
  for (let i = 0; i < nums.length; i++) {
    if (!map.has(nums[i])) {
      map.set(nums[i], 1);
    } else {
      map.set(nums[i], map.get(nums[i]) + 1);
    }
  }
  for (const [key, value] of map) {
    if (value === 1) {
      ans += key;
    }
  }
  return ans;
};
